test/object-store/c_api/c_api.cpp

```diff
@@ -4746,6 +4746,19 @@ TEST_CASE("app: flx-sync basic tests", "[c_api][flx][sync]") {
                            {"_id", bar_obj_id}, {"name", std::string{"bar"}}, {"value", static_cast<int64_t>(10)}}));
     });
 
+    SECTION("fail to convert a flx sync realm") {
+
+        harness.do_with_new_realm([&](SharedRealm realm) {
+            realm_t c_wrap_realm(realm);
+            TestFile dest_test_file;
+            auto dest_config = make_config(dest_test_file.path.c_str(), false);
+            dest_config->schema_version = 0;
+            auto res = realm_convert_with_config(&c_wrap_realm, dest_config.get(), false);
```

Comment:
do we want to check which exception gets thrown? if not, do we need a custom exception type for this or can you just throw a std::logic_error directly?

Reply:
This also doesn't seem like the right place to test this? This limitation isn't a C API thing, so it shouldn't be in the C API tests.

Reply:
Probably could do it with a simple std::logic_error.